src/nrc/api/tasks.py

```diff
@@ -1,41 +1,52 @@
 import json
-import logging
 
 from django.conf import settings
 from django.core.management import call_command
 from django.core.serializers.json import DjangoJSONEncoder
 from django.utils.translation import gettext_lazy as _
 
 import requests
+import structlog
 from notifications_api_common.autoretry import add_autoretry_behaviour
+from structlog.contextvars import bind_contextvars
 
 from nrc.celery import app
 from nrc.datamodel.models import Abonnement, NotificatieResponse
 
-logger = logging.getLogger(__name__)
+from .types import SendNotificationTaskKwargs
+
+logger = structlog.stdlib.get_logger(__name__)
 
 
 class NotificationException(Exception):
     pass
 
 
-@app.task
-def deliver_message(sub_id: int, msg: dict, **kwargs) -> None:
+@app.task(bind=True)
+def deliver_message(
+    self, sub_id: int, msg: SendNotificationTaskKwargs, **kwargs
+) -> None:
     """
     send msg to subscriber
 
     The delivery-result is logged in "NotificatieResponse"
     """
     notificatie_id: int = kwargs.pop("notificatie_id", None)
 
+    # `autoretry_attempt_number` is the number of times the same task was automatically retried
+    # `attempt_number` is the number of tasks that were started for this notification (without counting automatic retries)
+    autoretry_attempt_number = self.request.retries + 1
+    attempt_number = kwargs.get("attempt", 1)
```

Comment:
Not really sure what a proper name for each of these is, for instance if I have autoretry configured to do 3 retries and I attempt to send a notification via the API, `attempt_number` will be 1 for each of the times this task is retried and `autoretry_attempt_number` will be incremented accordingly. If I later retry this notification manually via the admin, `attempt_number` will be incremented to 2, and `autoretry_attempt_number` will start from 1 again and be incremented per retry.